test/brsTypes/components/RoSGNode.test.js

```diff
@@ -278,4 +278,333 @@ describe("RoSGNode", () => {
+
+                // get last 2 children
+                let result = getChildren.call(interpreter, new Int32(2), new Int32(2));
+                let childCount = getChildCount.call(interpreter);
+                expect(childCount).toEqual(new Int32(4));
+                expect(result).toBeInstanceOf(RoArray);
+                expect(result.elements).toEqual([child3, child4]);
+
+                // pass in more children count than there are children, should just return 2 with offset of 2
+                let result1 = getChildren.call(interpreter, new Int32(20), new Int32(2));
+                expect(result1).toBeInstanceOf(RoArray);
+                expect(result1.elements).toEqual([child3, child4]);
+
+                // pass in negative indices
+                let result2 = getChildren.call(interpreter, new Int32(-10), new Int32(-50));
+                expect(result2).toBeInstanceOf(RoArray);
+                expect(result2.elements).toEqual([]);
+
+                // pass in just negative start index
+                let result3 = getChildren.call(interpreter, new Int32(5), new Int32(-50));
+                expect(result3).toBeInstanceOf(RoArray);
+                expect(result3.elements).toEqual([]);
+
+                // get first 3 children
+                let result4 = getChildren.call(interpreter, new Int32(3), new Int32(0));
+                expect(result4).toBeInstanceOf(RoArray);
+                expect(result4.elements).toEqual([child1, child2, child3]);
+            });
+        });
+
+        describe("removechild", () => {
+            it("remove a child node from parent", () => {
+                let parent = new RoSGNode([
+                    { name: new BrsString("parent"), value: new BrsString("1") },
+                ]);
+                let child1 = new RoSGNode([
+                    { name: new BrsString("child"), value: new BrsString("2") },
+                ]);
+                let child2 = new RoSGNode([
+                    { name: new BrsString("child"), value: new BrsString("3") },
+                ]);
+                let child3 = new RoSGNode([
+                    { name: new BrsString("child"), value: new BrsString("4") },
+                ]);
+
+                let removeChild = parent.getMethod("removechild");
+                let getChildren = parent.getMethod("getchildren");
+                let getChildCount = parent.getMethod("getchildcount");
+                let appendChild = parent.getMethod("appendchild");
+
+                appendChild.call(interpreter, child1);
+                appendChild.call(interpreter, child2);
+                appendChild.call(interpreter, child3);
+
+                // remove child 2
+                removeChild.call(interpreter, child2);
+                let result = getChildren.call(interpreter, new Int32(-1), new Int32(0));
+                let childCount = getChildCount.call(interpreter);
+                expect(removeChild).toBeTruthy();
```

Comment:
I was only using `.toBeTruth()` to ensure the method existed before I called it.  Since you've already called `removeChild` by this line, there's no need to check it 😛

Reply:
let me move the check a bit upwards so it makes more sense